shell/scripts/test-plugins-build.sh

```diff
@@ -73,17 +73,20 @@ export YARN_REGISTRY=$VERDACCIO_YARN_REGISTRY
 export NUXT_TELEMETRY_DISABLED=1
 
 # Remove test package from previous run, if present
-rm -rf ${BASE_DIR}/pkg/test-pkg
+if [ $TEST_PERSIST_BUILD != "true" ]; then
+  echo "Removing folder ${BASE_DIR}/pkg/test-pkg"
+  rm -rf ${BASE_DIR}/pkg/test-pkg
+fi
 
 # We need to patch the version number of the shell, otherwise if we are running
 # with the currently published version, things will fail as those versions
 # are already published and Verdaccio will check, since it is a read-through cache
-sed -i.bak -e "s/\"version\": \"[0-9]*.[0-9]*.[0-9]*\",/\"version\": \"${SHELL_VERSION}\",/g" ${SHELL_DIR}/package.json
+sed -i.bak -e "s/\"version\": \"[0-9]*.[0-9]*.[0-9]*\(-alpha\.[0-9]*\|-release[0-9]*.[0-9]*.[0-9]*\|-rc\.[0-9]*\)\{0,1\}\",/\"version\": \"${SHELL_VERSION}\",/g" ${SHELL_DIR}/package.json
```

Comment:
comment only. at some point we should just use the suggested semver regex https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string. if this is tested and works though, no need to go through another round